scripts/ci/build_push_images.sh

```diff
@@ -1,51 +0,0 @@
-#!/bin/bash -xe
```

Comment:
could we keep this file, and call it like this?
```
docker.withRegistry(...) {
    sh './scripts/ci/build_push_images.sh ...'
}
```
and if not, then could we at least call a `build_push_images.groovy` script?

Reply:
why keep it if we don't use it?

Reply:
we would use it in my 1st suggestion (as can be seen in the `sh` line).
both options I raised have these benefits:
1. reduces PR diff size (decreasing the chance of mistakes, e.g. [losing checks](https://github.com/IBM/ibm-block-csi-driver/pull/245#discussion_r507626972))
2. keeps logic outside the pipeline (makes it all more modular, reusable and in smaller, more manageable files).

the 1st option also has this benefit:
3. keeps logic in bash (a native cross-platform language that people are more familiar with - easier to maintain and more portable).

now that I answered your "why", could we go back to my question - could we do this?

Reply:
I don't think that we should, when you are using Jenkinsfile you want to use the Jenkins plugins. 
When there is an option to use a Jenkins plugin, it's better to use it, because it gives you the option of better error handling and in my opinion more readable code, the plugin's syntax is pretty simple on most of the time and more readable.
Also, I think that we should avoid using shell scripts in Jenkinsfile instead of plugins for the same reason that you try to avoid shell module in Ansible and try to use builtin modules if you can.

Reply:
you only (partially) addressed the 3rd benefit (and thus only to my 1st option) - please address as much options and arguments as possible.

why would the plugin give better error handling? could you elaborate?

I think you are ignoring the limitations of the plugin's "simple" syntax - most of our logic cannot be used with it (env vars validation, sed, saving a tags file, etc...) and so our logic becomes separated into different languages and places, coupled to jenkins, with no real added value.